internal/resources/diffValidation/resource_instances.go

```diff
@@ -204,3 +208,17 @@ func (i *Instance) instanceValidatePowerTransition() error {
 
 	return fmt.Errorf("power operation not allowed from %s state to %s state", oldPowerStr, newPowerStr)
 }
+
```

Comment:
Please check the following scenario here.
1. Create a VM with a single snapshot and try to delete the snapshot. I hope this gets handled here.
2. Create a VM with single snapshot and add one more snapshot and try delete the second snapshot. I doubt if this case is handled here.
3. Create a VM with the single snapshot and add one more snapshot and try to delete both the snapshots.
4. Create a VM without any snapshot and call update again on the same ( to make sure your changes are not affecting other usecases.

Reply:
had offline discussion on these. 1 is handled. 2 & 3 may not be applicable as the tfstate file has the info about latest snapshot only. 4 is taken care